internal/contact/email.go

```diff
@@ -0,0 +1,43 @@
+package contact
+
+import (
+	"context"
+	"fmt"
+	"github.com/CollActionteam/collaction_backend/internal/constants"
+	"github.com/CollActionteam/collaction_backend/internal/models"
+)
+
+const separator = "### app version:"
+
+type EmailRepository interface {
+	Send(ctx context.Context, data models.EmailData) error
+}
+
+type ConfigManager interface {
+	GetParameter(name string) (string, error)
+}
+
+type Contact struct {
+	EmailRepository EmailRepository
+	ConfigManager   ConfigManager
+	Stage           string
+}
+
+func NewContactService(emailRepository EmailRepository, configManager ConfigManager, stage string) *Contact {
```

Comment:
I think that here should return an interface instead of a struct.

Reply:
by returning a struct, everyone can create their required interface and use what they want from this logic.
if you check the example HTTP controller I created an interface and add the requirement that this handler needs to work.
and if we create a big interface with all methods of the service, it's make our abstraction weak, like go Proverbs `The bigger the interface, the weaker the abstraction.`

Reply:
also, we can move the interface created inside the HTTP controller and I think it's ok.

Reply:
NewContactService return a Contact. But I don't need to call NewContactService to create an instance of Contact.
I can do that 

```suggestion
c := Contact{} 
and if I  call c.SendEmail() it doesn't work because I don't pass the EmailRepository and ConfigManager parameters
```

what do you think of this kind of approach?


```suggestion
// struct private 
type contact struct {
	EmailRepository EmailRepository
	ConfigManager   ConfigManager
	Stage           string
}

// All contactService methods in this interface
type ContactService interface {
	SendEmail(ctx context.Context, data models.EmailContactRequest) error
}
// I only can get a new instance of contactService call this method
func NewContactService(emailRepository EmailRepository, configManager ConfigManager, stage string) ContactService {
	return &contact{EmailRepository: emailRepository, ConfigManager: configManager, Stage: stage}
}


func (e *contact) SendEmail(ctx context.Context, data models.EmailContactRequest) error {
	//return nil
}

```
 With this approach, I can easily mock the contactService behavior, just implementing the ContactServiceInterface

Reply:
yes and this is the user's responsibility (controller), we defined the contact requirements, and if anyone wants to use it, must provide it first.


with both ways, we can mock it easily and it will not prevent us from not testing it.
I agree with this way as well.
I don't see any reason why not to use the first approach but we can do your approach and it will not make to difference.

and if we not making the `Contact` fields public.

```
// struct private 
type contact struct {
	emailRepository EmailRepository
	configManager   ConfigManager
	stage           string
}
```